module/Photo/src/Photo/Service/Admin.php

```diff
@@ -178,7 +178,8 @@ public function upload($files, $album)
             }
         } else {
             throw new \Exception(
-                $translator->translate('The uploaded file is not a valid image')
+                $translator->translate("The uploaded file is not a valid image \nError: " .
+                    json_encode($imageValidator->getMessages()))
```

Comment:
Implode pretty much does the same, except for it results in a slightly more compact message. Keeping the message as small as possible is good in this case, since there is very little room in the view to display it.